src/context/DataContext/DataReducer.js

```diff
@@ -0,0 +1,66 @@
+export const initialState = {
+  videos: [],
+  categories: [],
+  watchLater: [],
+  likedVideos: [],
+};
+
+export const filterReducer = (state, action) => {
+  switch (action.type) {
+    case "SORT":
+      return {
+        ...state,
+        sort: action.payload,
+      };
+
+    case "CATEGORIES":
+      return {
+        ...state,
+        category: action.payload.isDel ? "" : action.payload.categoryName,
+      };
+    default:
+      return state;
+  }
+};
+
+export const reducer = (state, action) => {
+  switch (action.type) {
```

Comment:
You can destructure const {payload, type} = action

Reply:
ok will do that.